src/transformers/modeling_tf_utils.py

```diff
@@ -196,10 +196,13 @@ def hf_compute_loss(self, labels, logits):
             from_logits=True, reduction=tf.keras.losses.Reduction.NONE
         )
         # make sure only labels that are not equal to -100 affect the loss
-        active_loss = tf.not_equal(tf.reshape(labels, (-1,)), -100)
-        reduced_logits = tf.boolean_mask(tf.reshape(logits, (-1, shape_list(logits)[2])), active_loss)
-        labels = tf.boolean_mask(tf.reshape(labels, (-1,)), active_loss)
-        return loss_fn(labels, reduced_logits)
+        unmasked_loss = loss_fn(labels, logits)
+        loss_mask = tf.cast(labels != -100, dtype=unmasked_loss.dtype)
+        loss_denominator = tf.reduce_sum(loss_mask, axis=1)
+        # Masked positions will have a loss of NaN because -100 is not a valid label
+        masked_loss = tf.math.multiply_no_nan(unmasked_loss, loss_mask)
+        reduced_masked_loss = tf.reduce_sum(masked_loss, axis=1) / loss_denominator
+        return reduced_masked_loss
```

Comment:
I was checking the docs now, and PT only outputs one number for the batch -- so it makes sense to include the sum here 👍 

(we would have to update the TF docstring)

Reply:
(BTW, the PT output the average instead of the sum, as I see in `GPT2LMHeadModel` - `CrossEntropyLoss` have `reduction='mean'` by default).

Reply:
It will be impossible to keep the old behaviour with XLA, because the number of 'active' tokens will change in each batch.

We could return a scalar number, but in Keras it's nice to return a vector of per-sample losses, because this means the user can use the `sample_weight` argument to `fit()` if they want to. I think that's fairly uncommon though, so if we want to stick with a scalar, that's fine!